[#24] Implement room reservation based on dates

Zakhar23 opened the pull request:
[#24] Implement room reservation based on dates
[#25] Implement cancellation of room reservation
 + implement check room reservation

Zakhar23 committed:
[#24] Implement room reservation based on dates
[#25] Implement cancellation of room reservation
 + implement check room reservation

Zakhar23 committed:
[#24] Implement room reservation based on dates
[#25] Implement cancellation of room reservation
 + implement check room reservation

Sheptytskyid commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,54 @@
+    public List<Room> checkRoomReservation(Date reservedFrom, Date reservedTo, List<Room> findRooms) {
+        List<Room> toDel = new ArrayList<>();
+        for (Room findRoom : findRooms) {
+            if (reservationDao.getAll().stream().filter(p -> p.getReservedRoom().equals(findRoom)
```
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (10) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1067)

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -7,9 +7,12 @@
     private Date reservedTo;
     private User reservedUser;
+    private Room reservedRoom;
+    private Hotel reservedHotel;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "reservedHotel" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented:
SonarQube analysis reported 6 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -7,9 +7,12 @@
     private Date reservedFrom;
     private Date reservedTo;
     private User reservedUser;
+    private Room reservedRoom;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "reservedRoom" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/com/manardenza/dao/ReservationDaoImpl.java:
```diff
@@ -0,0 +1,31 @@
+package com.manardenza.dao;
+
+import com.manardenza.entity.Hotel;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.manardenza.entity.Hotel'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Sheptytskyid commented on src/main/java/com/manardenza/dao/ReservationDaoImpl.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    @Override
+    public Reservation save(Reservation obj) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this method to simply inherit it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1185)

KoliushkoArtem replied:
Delete this override

Sheptytskyid commented on src/main/java/com/manardenza/dao/ReservationDaoImpl.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    @Override
+    public List<Reservation> getAll() {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this method to simply inherit it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1185)

KoliushkoArtem replied:
Delete this override

Zakhar23 committed:
[#24] Implement room reservation based on dates
[#25] Implement cancellation of room reservation
 + implement check room reservation

Sheptytskyid commented:
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Zakhar23 committed:
[#24] Implement room reservation based on dates
[#25] Implement cancellation of room reservation
 + implement check room reservation

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -7,9 +7,31 @@
+    public boolean newReservedOk(Date newReservedFrom, Date newReservedTo) {
+        if (newReservedFrom.after(reservedFrom) && newReservedFrom.before(reservedTo)) {
+            return false;
+        } else if (newReservedTo.after(reservedFrom) && newReservedTo.before(reservedTo)) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 19. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -7,9 +7,31 @@
+            return false;
+        } else if (newReservedTo.after(reservedFrom) && newReservedTo.before(reservedTo)) {
+            return false;
+        } else if (newReservedFrom.before(reservedFrom) && newReservedTo.after(reservedTo)) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 21. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -7,9 +7,31 @@
+            return false;
+        } else if (newReservedFrom.before(reservedFrom) && newReservedTo.after(reservedTo)) {
+            return false;
+        } else if (newReservedFrom.equals(reservedFrom)) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 23. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)

Sheptytskyid commented:
SonarQube analysis reported 9 issues
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -7,9 +7,31 @@
+            return false;
+        } else if (newReservedFrom.equals(reservedFrom)) {
+            return false;
+        } else if (newReservedFrom.equals(reservedTo)) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 25. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -7,9 +7,31 @@
+            return false;
+        } else if (newReservedFrom.equals(reservedTo)) {
+            return false;
+        } else if (newReservedTo.equals(reservedFrom)) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 27. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -7,9 +7,31 @@
+            return false;
+        } else if (newReservedTo.equals(reservedFrom)) {
+            return false;
+        } else if (newReservedTo.equals(reservedTo)) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') This branch's code block is the same as the block for the branch on line 29. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1126)

Zakhar23 committed:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6 into 24_Implement_room_reservation_based_on_dates

Zakhar23 committed:
Merge branch 'master' into 24_Implement_room_reservation_based_on_dates

Zakhar23 committed:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6 into 24_Implement_room_reservation_based_on_dates

# Conflicts:
#	src/main/java/com/manardenza/entity/Reservation.java

Zakhar23 committed:
Merge remote-tracking branch 'origin/24_Implement_room_reservation_based_on_dates' into 24_Implement_room_reservation_based_on_dates

Zakhar23 committed:
24,25

Zakhar23 committed:
24,25 fix checkstyle

Zakhar23 committed:
24, 25

Zakhar23 committed:
Merge branch 'master' into 24_Implement_room_reservation_based_on_dates

Sheptytskyid commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,60 @@
+
+    public Map<String, List<Room>> checkRoomReservation(Date reservedFrom, Date reservedTo,
+                                                        Map<String, List<Room>> rooms) {
+        for (String key : rooms.keySet()) {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2864)

Sheptytskyid commented:
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DatabaseSeeder.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/26fbd578543c01eb0374c98687ba8e1cf18f5855/src/main/java/com/manardenza/utils/DatabaseSeeder.java#L32): Define a constant instead of duplicating this literal "Vasya" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Zakhar23 committed:
24, 25 entry

Zakhar23 committed:
Merge remote-tracking branch 'origin/24_Implement_room_reservation_based_on_dates' into 24_Implement_room_reservation_based_on_dates

Sheptytskyid commented:
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [DatabaseSeeder.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/9621008815dd97ba01def8f20ab9ac7405455520/src/main/java/com/manardenza/utils/DatabaseSeeder.java#L32): Define a constant instead of duplicating this literal "Vasya" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'

Zakhar23 committed:
24, 25 clean data

Sheptytskyid commented on src/main/java/com/manardenza/utils/DatabaseSeeder.java:
```diff
@@ -1,12 +1,15 @@
+import com.manardenza.dao.ReservationDaoImpl;
 import com.manardenza.dao.UserDaoImpl;
 import com.manardenza.entity.Hotel;
+import com.manardenza.entity.Reservation;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.manardenza.entity.Reservation'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Sheptytskyid commented on src/main/java/com/manardenza/utils/DatabaseSeeder.java:
```diff
@@ -21,6 +24,7 @@
     private static final String REIKARTZ = "Reikartz";
     private static final UserDaoImpl USERDAOIMPL = UserDaoImpl.getInstance();
     private static final HotelDaoImpl HOTELDAOIMPL = HotelDaoImpl.getInstance();
+    private static final ReservationDaoImpl RESERVATIONDAOIMPL = ReservationDaoImpl.getInstance();
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "RESERVATIONDAOIMPL" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented on src/main/java/com/manardenza/utils/DatabaseSeeder.java:
```diff
@@ -1,12 +1,15 @@
 import com.manardenza.entity.User;
 
 import java.util.ArrayList;
+import java.util.Date;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'java.util.Date'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Sheptytskyid commented:
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Zakhar23 committed:
24, 25

Sheptytskyid commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,60 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'java.util.stream.Collectors'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Sheptytskyid commented:
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Zakhar23 committed:
24, 25

Sheptytskyid commented:
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Zakhar23 committed:
24, 25

Zakhar23 committed:
new method getUserUserReservations

Zakhar23 committed:
new method getUserUserReservations

Sheptytskyid commented:
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Sheptytskyid commented on src/main/java/com/manardenza/controller/ReservationController.java:
```diff
@@ -0,0 +1,43 @@
+        return reservationService.checkRoomReservation(reservedFrom, reservedTo, rooms);
+    }
+
+    private List<Reservation> getAllUserReservations() {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "getAllUserReservations" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

zozich commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -3,14 +3,31 @@
+    private Room reservedRoom;
+    private Hotel reservedHotel;
+
+    public boolean overlaps(Date newReservedFrom, Date newReservedTo) {
```
This implementation is error prone. You may use YodaTime Interval to check this, or just simplify the logic. Two ranges overlap if:
(StartDate1 <= EndDate2) && (StartDate2 <= EndDate1)

zozich commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,69 @@
+        return reservationDao.save(reservation).getId();
+    }
+
+    public boolean cancelReservation(long idReservation) {
```
It may happen that this operation will not be stored. To be rewritten

zozich commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,69 @@
+
+    private List<Room> selectAvailableRooms(Date reservedFrom, Date reservedTo, List<Room> rooms) {
+        List<Room> roomsToDelete = new ArrayList<>();
+        for (Room room : rooms) {
```
Method can be optimized to avoid double loop (using `contains()`)

zozich reviewed (changes requested):
Please amend

Zakhar23 committed:
change method bookRoom

Zakhar23 committed:
change method selectAvailableRooms - avoid double loop
delete method overlaps

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -3,14 +3,16 @@
     private Date reservedTo;
     private User reservedUser;
+    private Room reservedRoom;
+    private Hotel reservedHotel;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "reservedHotel" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Sheptytskyid commented:
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Sheptytskyid commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -3,14 +3,16 @@
     private Date reservedFrom;
     private Date reservedTo;
     private User reservedUser;
+    private Room reservedRoom;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "reservedRoom" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Zakhar23 committed:
with Denis changes

Zakhar23 committed:
change code

Sheptytskyid commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,57 @@
+
+public final class ReservationService {
+
+    private static ReservationService instance;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "instance" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

zozich replied:
Remove it

Sheptytskyid commented:
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Sheptytskyid commented on src/main/java/com/manardenza/utils/Injector.java:
```diff
@@ -14,14 +17,19 @@
+    private static ReservationService reservationService = new ReservationService(reservationDao);
     private static HotelController hotelController = new HotelController(hotelService);
     private static UserController userController = new UserController(userService);
+    private static ReservationController reservationController = new ReservationController(reservationService);
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "reservationController" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

zozich commented on src/main/java/com/manardenza/entity/Reservation.java:
```diff
@@ -8,9 +8,11 @@
 @Data
 @AllArgsConstructor
-public class Reservation implements Serializable {
+public class Reservation extends AbstractObject implements Serializable {
```
AbstractObject already implements Serializable, so you can remove it from here

zozich commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,57 @@
+
+    private static ReservationService instance;
+    private ReservationDaoImpl reservationDao;
+    private CurrentUser currentUser;
```
This should also be injected via constructor

zozich commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    public boolean cancelReservation(long idReservation) {
+        return reservationDao.getAll().removeIf(p -> p.getId() == idReservation);
```
You cannot do like this - it will not be persisted. You should use delete() method

zozich commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    private List<Room> selectAvailableRooms(Date reservedFrom, Date reservedTo, List<Room> rooms) {
+        return rooms.stream().filter(room -> reservationDao.getAll().stream()
```
Loop inside loop. I asked to simplify this

zozich commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,57 @@
+    private List<Room> selectAvailableRooms(Date reservedFrom, Date reservedTo, List<Room> rooms) {
+        return rooms.stream().filter(room -> reservationDao.getAll().stream()
+                .filter(reservation -> reservation.getReservedRoom().equals(room))
+                .filter(reservation -> reservation.getReservedFrom().getTime() <= reservedTo.getTime()
```
Leave this as a method inside Reservation class - as it was previously

Zakhar23 committed:
corrected comments

Sheptytskyid commented:
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [Reservation.java#L6](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/26421372c8db65ece56fe9e874912acae21491f2/src/main/java/com/manardenza/entity/Reservation.java#L6): Remove this unused import 'java.io.Serializable'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Sheptytskyid commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,61 @@
+    private CurrentUser currentUser;
+
+    public ReservationService(ReservationDaoImpl reservationDao) {
+        this.reservationDao = reservationDao;
```
+currentUser = CurrentUser.getInstance();

Sheptytskyid commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,61 @@
+    public Long bookRoom(Date reservedFrom, Date reservedTo, Room room, Hotel hotel) {
+        Reservation newReservation = new Reservation(reservedFrom, reservedTo, currentUser.getUser(), room, hotel);
+        reservationDao.save(newReservation);
+
```
remove blank line

Sheptytskyid commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,61 @@
+
+    public boolean cancelReservation(long idReservation) {
+        if (reservationDao.getAll().stream().anyMatch(p -> p.getId() == idReservation)) {
+            reservationDao.delete(reservationDao.getAll().stream()
```
simplify: 
return: reservationDao.delete(reservationDao.getAll().stream().filter(p -> p.getId() == idReservation).findFirst().get());

Sheptytskyid commented on src/main/java/com/manardenza/service/ReservationService.java:
```diff
@@ -0,0 +1,61 @@
+                .filter(reservation -> reservation.getReservedUser().equals(currentUser.getUser()))
+                .collect(Collectors.toList());
+    }
+
```
remove blank line

KoliushkoArtem committed:
Added Denis and Andrew comments

Sheptytskyid commented:
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [Reservation.java#L6](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/b458c10bdb6d617696a8556c916c5549c66867ca/src/main/java/com/manardenza/entity/Reservation.java#L6): Remove this unused import 'java.io.Serializable'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

zozich reviewed (approved):
Ok
